app/src/main/java/app/revanced/integrations/patches/ReturnYouTubeDislikePatch.java

```diff
@@ -32,19 +32,17 @@ public static void newVideoLoaded(String videoId) {
      *
      * Required to update the UI after the user dislikes.
      *
-     * @param textRef Reference to the dislike char sequence.
+     * @param textRef Reference to the dislike char sequence. The CharSequence inside can be null.
      */
     public static void onComponentCreated(@NonNull Object conversionContext, @NonNull AtomicReference<CharSequence> textRef) {
         try {
             if (!SettingsEnum.RYD_ENABLED.getBoolean()) {
                 return;
             }
-            CharSequence original = textRef.get(); // CharSequence can be null
-            if (original instanceof Spanned) {
-                SpannableString replacement = ReturnYouTubeDislike.getDislikeSpanForContext(conversionContext, (Spanned) original);
-                if (replacement != null) {
-                    textRef.set(replacement);
-                }
+
+            SpannableString replacement = ReturnYouTubeDislike.getDislikeSpanForContext(conversionContext, textRef.get());
+            if (replacement != null) {
+                textRef.set(replacement);
             }
```

Comment:
From the given context it is not possible to assume that textRef.get() returns null which obfuscates the check here a bit. Though for now this should suffice.